### PR TITLE
feat: warn if app is started without booting

### PR DESCRIPTION
**Pull request by hacksparrow**
Warns if app is started without booting. Ref: https://github.com/strongloop/loopback-next/issues/6136.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/boot/src/mixins/boot.mixin.ts [hunk 1]
```diff
@@ -71,6 +71,23 @@ export function BootMixin<T extends MixinTarget<Application>>(superClass: T) {
...
+
+    /**
+     * Override to detect and warn about starting without booting.
+     */
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    public async start(): Promise<void> {
+      await super.start();
+      if (!this.booted) {
+        console.warn(
```
Comment (raymondfeng) [inline packages/boot/src/mixins/boot.mixin.ts:84]: I wish we have a better way for warnings. `console.warn` is not very flexible.

Comment (raymondfeng) [reply packages/boot/src/mixins/boot.mixin.ts:84]: Let's use `process.emitWarning()` - https://nodejs.org/api/process.html#process_process_emitwarning_warning_type_code_ctor

Comment (hacksparrow) [reply packages/boot/src/mixins/boot.mixin.ts:84]: Wouldn't the warning be lost if they are not being actively listened for? If it requires an additional step for the developer to be able to be warned of problems, it is not en effective solution. The `console.log()` method is also used by [datasource](https://github.com/strongloop/loopback-next/blob/master/packages/boot/src/booters/datasource.booter.ts#L56) and [repository](https://github.com/strongloop/loopback-next/blob/master/packages/boot/src/booters/repository.booter.ts#L52) booters.

If `console.log()` is not flexible enough, we should be come up with a solution for it instead of passing on the pain to the user.

Comment (raymondfeng) [reply packages/boot/src/mixins/boot.mixin.ts:84]: `process.emitWarning(...)` will print the message to `stderr` as I tested:

```
process.emitWarning('xyz');
> (node:85982) Warning: xyz
```

Comment (hacksparrow) [reply packages/boot/src/mixins/boot.mixin.ts]: I see, updated.

File: packages/boot/src/mixins/boot.mixin.ts [hunk 2]
```diff
@@ -71,6 +65,23 @@ export function BootMixin<T extends MixinTarget<Application>>(superClass: T) {
...
+     */
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    public async start(): Promise<void> {
+      await super.start();
+      if (!this.booted) {
+        process.emitWarning(
+          'App started without booting. Did you forget to call ' +
+            '`await app.boot()`?',
+        );
```
Comment (raymondfeng) [inline packages/boot/src/mixins/boot.mixin.ts:81]: Let's pass in `type` (such as 'LoopBackWarning') in `options` so that the printout will have `LoopBackWarning` prefix.

File: packages/boot/src/mixins/boot.mixin.ts [hunk 3]
```diff
@@ -24,10 +22,6 @@ import {Bootstrapper} from '../bootstrapper';
 import {BootBindings, BootTags} from '../keys';
 import {Bootable, Booter, BootOptions, InstanceWithBooters} from '../types';
 
-// FIXME(rfeng): Workaround for https://github.com/microsoft/rushstack/pull/1867
-/* eslint-disable @typescript-eslint/no-unused-vars */
-import * as loopbackContext from '@loopback/core';
-import * as loopbackCore from '@loopback/core';
```
Comment (raymondfeng) [inline packages/boot/src/mixins/boot.mixin.ts:30]: Please revert this change.
